src/chipid.rs

```diff
@@ -11,15 +11,32 @@ pub enum EmbeddedProcessor {
     CortexM4,
 }
 
-#[derive(Debug)]
-pub enum Architecture {
-    AtSam4E,
-    AtSam4SxA,
-    AtSam4SxB,
-    AtSam4SxC,
+#[derive(Clone, Copy, Debug)]
+pub enum Family {
+    AtSam4e,
+    AtSam4s,
 }
 
-#[derive(Debug)]
+#[derive(Clone, Copy, Debug)]
+pub enum Model {
+    AtSam4e8c,
+    AtSam4e8e,
+    AtSam4e16c,
+    AtSam4e16e,
+
+    AtSam4s8b,
```

Comment:
AtSam4s2a, AtSam4s2b, AtSam4s2c, AtSam4s4a, AtSam4s4b, AtSam4s4c?

Reply:
Yep, I missed those.   I was looking at the errata table instead of the one in the CHIPID section.   oops.